src/pages/settings/Payments/PaymentsPage/index.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import {View} from 'react-native';
+import PaymentMethodList from '../PaymentMethodList';
+import ROUTES from '../../../../ROUTES';
+import HeaderWithCloseButton from '../../../../components/HeaderWithCloseButton';
+import ScreenWrapper from '../../../../components/ScreenWrapper';
+import Navigation from '../../../../libs/Navigation/Navigation';
+import styles from '../../../../styles/styles';
+import withLocalize, {withLocalizePropTypes} from '../../../../components/withLocalize';
+import compose from '../../../../libs/compose';
+import Button from '../../../../components/Button';
+import KeyboardAvoidingView from '../../../../components/KeyboardAvoidingView';
+import FixedFooter from '../../../../components/FixedFooter';
+import getPaymentMethods from '../../../../libs/actions/PaymentMethods';
+import Popover from '../../../../components/Popover';
+import {PayPal} from '../../../../components/Icon/Expensicons';
+import MenuItem from '../../../../components/MenuItem';
+
+const propTypes = {
+    ...withLocalizePropTypes,
+};
+
+class PaymentsPage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.PAYPAL = 'payPalMe';
+
+        this.state = {
+            showAddPaymentMenu: false,
+            anchorPositionTop: 0,
+            anchorPositionLeft: 0,
+        };
+
+        this.paymentMethodPressed = this.paymentMethodPressed.bind(this);
+        this.addPaymentMethodTypePressed = this.addPaymentMethodTypePressed.bind(this);
+        this.hideAddPaymentMenu = this.hideAddPaymentMenu.bind(this);
+    }
+
+    componentDidMount() {
+        getPaymentMethods();
```

Comment:
So I think the way this works in your current implementation is:

1. The payment methods are stored in Onyx.
1. When the payments page opens the saved ones will appear more-or-less immediately, and we query the API for updated payment methods
1. If another payment was added, then it will appear when the API request finishes.

So NAB (or can be done in a follow-up), but it might be good to add a loading indicator in the process:

1. The payment methods are stored in Onyx
1. When the payment page opens, the saved ones will appear more-or-less immediately. If we are online we query the API for updated payment methods and display a loading indicator.
1. When the API request finishes, the loading indicator hides and any new payment methods are displayed.

Reply:
I think that makes sense but like you said it's kinda a QOL improvement and I'd rather get this out ASAP to unblock the rest of the project :)